okhttp/src/main/kotlin/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -368,6 +384,18 @@ class RealConnection(
     }
   }
 
+  private fun getMaximumAllocationLimit(): Int {
+    // if we have not negotiated a max per streams yet, don't check for the policy override
+    val negotiatedMaxCurrentStreams = lastMaxConcurrentStreamsFromSettings ?: return 1
```

Comment:
Isn't this just defaulted to 1 for Http/1 and Http2Connection.DEFAULT_SETTINGS.getMaxConcurrentStreams() for Http/2?

should it be a lateinit and start sets to one or the other?

Reply:
maybe?

`private var lastMaxConcurrentStreamsFromSettings = 1` would avoid the early return, but then we're checking against the policy.

Would it be better to have the null and early return or check the policy every time, even if we're not in HTTP/2 land?